logger/src/main/java/com/orhanobut/logger/FileLogger.java

```diff
@@ -0,0 +1,362 @@
+ *             .logLevel(android.util.Log.INFO) // only log INFO and up
+ *             .build())));
+ * }
+ * </pre>
+ * Alternatively <code>.folderFullPath(path)</code> can be used to log on internal memory
+ */
+public class FileLogger implements LogAdapter {
+
+    //region static definitions
+    private static final String BEGIN = TOP_BORDER;
+    private static final String MID = MIDDLE_BORDER;
+    private static final String END = BOTTOM_BORDER;
+    private static final String HORZ_LINE = String.valueOf(HORIZONTAL_DOUBLE_LINE);
+    private static final String CSV_SEPARATOR = ",";
+    private static final String NEW_LINE = System.getProperty("line.separator");
+    private static final String FILE_SEPARATOR = File.separator;
+    // I'm using br here because it's easy to find search, and won't be confused easily
+    private static final String NEW_LINE_REPLACEMENT = " <br> ";
+    private static final int DEFAULT_MAX_LINES = 2048;
+    private static final List<String> MARKERS = Arrays.asList(BEGIN, MID, END);
+
+    private static final SparseArray<String> LEVELS;
+
+    static {
+        LEVELS = new SparseArray<>();
+        LEVELS.append(Log.VERBOSE, "VERBOSE");
+        LEVELS.append(Log.DEBUG, "DEBUG");
+        LEVELS.append(Log.INFO, "INFO");
+        LEVELS.append(Log.WARN, "WARN");
+        LEVELS.append(Log.ERROR, "ERROR");
+        LEVELS.append(Log.ASSERT, "ASSERT");
+    }
+    //endregion
+
+    private final Handler handler;
+    private final StringBuilder stringBuilder = new StringBuilder(200);
+    private final SimpleDateFormat dateFormat;
+    private final Date date = new Date();
+    private final String separator;
+    private final int logLevel;
+
+    private FileLogger(int logLevel, int maxLines, String folder, String separator,
+                       String dateFormat, Locale locale, Looper looper) {
+        this.logLevel = logLevel;
+        this.separator = separator;
+        this.dateFormat = new SimpleDateFormat(dateFormat, locale);
+        this.handler = new WriteHandler(looper, folder, maxLines);
+    }
+
+    @Override
+    public void d(String tag, String message) {
+        log(Log.DEBUG, tag, message);
+    }
+
+    @Override
+    public void e(String tag, String message) {
+        log(Log.ERROR, tag, message);
+    }
+
+    @Override
+    public void w(String tag, String message) {
+        log(Log.WARN, tag, message);
+    }
+
+    @Override
+    public void i(String tag, String message) {
+        log(Log.INFO, tag, message);
+    }
+
+    @Override
+    public void v(String tag, String message) {
+        log(Log.VERBOSE, tag, message);
+    }
+
+    @Override
+    public void wtf(String tag, String message) {
+        log(Log.ERROR, tag, message);
+    }
+
+    //region message builder
+    private synchronized void log(int level, String tag, String message) {
+        if (logLevel <= 0 || level < logLevel) return;
```

Comment:
My suggestion would be to add values to the current ENUM and use switch-case fallthrough. This has the added benefit to already consider a hierarchy between the logs. E.g. if you're logging INFO you also want to have WARN/ERROR.

But of-course bitwise masks works just the same. I'll wait for your comment on this one, as I'm OK with both.